### ISSUE TITLE
Learning 3D Human Pose Estimation from Dozens of Datasets using a Geometry-Aware Autoencoder to Bridge Between Skeleton Formats

### ISSUE THREAD

**Issue by gwleeee** [gwleeee/PaperReview#6]
# Learning 3D Human Pose Estimation from Dozens of Datasets using a Geometry-Aware Autoencoder to Bridge Between Skeleton Formats

paper url: https://arxiv.org/abs/2212.14474
WACV 2023

## Abstract
- 3D 휴먼 포즈 데이터셋들의 스켈레톤 포맷을 일치시키는 것은 매우 중요한 문제
- 스켈레톤별 별도의 헤드를 통해 포즈 추론시 깊이정보가 많이 차이남 -> 스켈레톤간 정보 공유가 불충분
- Affine-Combining AutoEncoder(ACAE) 제안 -> 랜드마크 차원축소
- 잠재 3D 키포인트는 스켈레톤들 사이의 중복성을 캡쳐 -> 정규화에 사용
- 28개의 3D 휴먼 포즈 데이터셋을 이용하여 하나의 모델 학습

<img width="1000" alt="image" src="https://github.com/gwleeee/PaperReview/assets/83268025/0186f51c-2b72-4c02-839b-45f15b620f83">

## Introduction
- 3D 휴먼 포즈 데이터는 협소한 스튜디오에서 제작하기 때문에 다양성이 적음
- 추론 성능 향상을 위해 가능한 많은 데이터셋을 이용해야 하지만, 다양한 데이터셋으로 단일 모델을 학습할 경우 성능에 대한 보장이 충분치 않음 (각 데이터셋 마다 스켈레톤의 정의가 다르기 때문)
- 과거에는 이러한 한계를 극복하기 위해, 룰 베이스의 스켈레톤 수정방법을 사용
- e.g. hip-pelvis 사이의 거리를 일치시키는 등

**Multi-skeleton 3D human pose estimation**
스켈레톤 정의가 서로 다른 수십개의 데이터셋을 한번에 학습하기 위해 어떻게 병합시킬까?

- 다중 데이터셋을 이용하여 그냥 학습하면, 평균적인 스켈레톤을 출력하여 성능이 저하됨!
- 하나의 백본을 이용하여 특징들을 학습하고, 각 데이터셋에 맞는 헤드로 스켈레톤을 출력하는 경우, 깊이 정보가 뒤죽박죽함 -> 백본에서 정보 공유가 제대로 되질 않고 있다!

**autoencoder-based  dimensionality  reduction  technique**
<img width="446" alt="image" src="https://github.com/gwleeee/PaperReview/assets/83268025/bcf38c9c-ad2a-4f1f-a239-cff6fa0dbeca">

모든 데이터셋의 joint를 잠재 키포인트셋으로 차원축소

**Main contributions**
1. 28개의 개별적인 3D 휴먼 포즈 데이터셋을 통합
2. 키포인트 차원축소 기법 ACAE 제안
3. ACAE를 이용하여 모델 예측을 정규화 하여 추론 품질을 높이고, 잠재 키포인트 또한 예측 가능하도록 함

<img width="500" alt="image" src="https://github.com/gwleeee/PaperReview/assets/83268025/e0146d0d-7c38-4f24-9423-8804117108e0">

## Method
<img width="1000" alt="image" src="https://github.com/gwleeee/PaperReview/assets/83268025/3c57303d-329b-41c1-af36-81cd1ad0834a">

### 1. Initial model training
- 모든 데이터셋의 joint를 추론할 수 있도록 출력 (마치 멀티헤드를 이용하여 각 데이터셋별로 스켈레톤을 추론하는 것처럼)
- 여러 스켈레톤의 joint 상관관계를 고려치 않고 냅다 학습

**Loss function**
<img width="393" alt="image" src="https://github.com/gwleeee/PaperReview/assets/83268025/6d58a1f3-8f1d-43e1-a63b-19097566a5e7">
- L_meanrel: 출력에 대한 L1 loss
- L_proj: 이미지로 투영된 2D 좌표에 대한 L1 loss
- L_abs: 카메라 좌표에서의 L1 loss
- 특정 스켈레톤 데이터가 학습 입력일때는 연관된 출력에 대해서만 loss 계산

### 2. Psudo-ground truth generation
- 신뢰도가 높은 데이터셋을 이용하여 가짜 GT를 생성 -> autoencoder를 학습하는데 사용
- H36M, MoVi 등의 데이터를 사용했음
- 이 데이터셋의 이미지를 initial model에 통과시켜 J개의 joint 출력을 가지는 벡터 K개를 만듦

### 3. Affine-combining autoencoder
<img width="341" alt="image" src="https://github.com/gwleeee/PaperReview/assets/83268025/cbc6077a-dca5-44d8-9d2f-33295fe7a9ce">

#### encoder
<img width="500" alt="image" src="https://github.com/gwleeee/PaperReview/assets/83268025/1b3d1723-e457-4fc9-a28b-f62dc9e0d7cf">

- 인코더는 affine 변환을 수행할 수 있는 단순한 weight의 조합으로 되어있음
- J개의 모든 joint를 가중합 하여 latent joint 하나를 생성
- J -> L 차원으로의 fully-connected 연산으로 볼수 있음 (단 한 노드에 대한 weight의 합은 1)
#### decoder
<img width="500" alt="image" src="https://github.com/gwleeee/PaperReview/assets/83268025/56ff5675-8642-4f21-9626-a17ed7a6168a">

- 디코더는 인코더의 반대의 과정이므로, L -> J 차원으로의 확장

#### loss function
<img width="500" alt="image" src="https://github.com/gwleeee/PaperReview/assets/83268025/8b5f92c7-46d9-468f-af76-fcb650c71d9e">

- L_reconstr는 입출력에 대한 L1 loss
- L_sparse는 정규화 강도를 조정

#### others
- Reconstruction loss on 2d projection: 가짜 GT는 2d에서 더 안정적이기 때문에, 2d 투영 결과로도 loss를 발생시킴
- Chirality equivariance: 사람은 좌우 대칭이라는 성질을 이용해서, 인코더-디코더의 weight matrix를 (좌-중앙-우)로 그룹화
- Head keypoint weighting: 얼굴영역 키포인트는 매우 중요하므로, 얼굴영역에 대해서는 loss를 10배정도 더 줌

### 4. Consistency fine-tuning
<img width="500" alt="image" src="https://github.com/gwleeee/PaperReview/assets/83268025/bbe26b32-7f81-4043-88c5-eeac9f2c5c98">

- autoencoder의 weight를 freeze하고 3D pose estimator를 학습

<img width="500" alt="image" src="https://github.com/gwleeee/PaperReview/assets/83268025/ade3f503-8b56-4161-990b-faff6b5d1169">

- 그림과 같이 잠재 키포인트를 3D pose estimator가 바로 생성하게 하는 방식과, student-teacher 방식을 이용하여 잠재 키포인트를 생성하도록 하는 방법도 있음

## Experiments and Results
- 베이스 모델: MeTRAbs (백본: EfficientNetV2-S)
- 학습
  1. 베이스 모델: 전체 데이터셋에 대해 400k step
  2. 오코인코더: EffV2-L 모델로 획득한 가짜 GT를 이용하여 학습 (head는 고정하고, 백본만 EffV2-L로 교체하여 출력하는건가?)
  3. 파인튜닝: 백본에 대해서 작은 lr로 40k
  4. 입력: 256x256 크롭 이미지

<img width="1000" alt="image" src="https://github.com/gwleeee/PaperReview/assets/83268025/50c113a7-5f10-450f-9151-d1c712ab6b42">

- 많은 데이터셋을 사용했을때 MuPoTS, 3DPW, MPI 데이터셋에서는 더욱 좋은 성능을 보여줬지만, H36M에선 적은 데이터로 학습시 좋은 성능을 보여줌
- in-the-wild 데이터셋에서 더 좋은 일반화 성능을 보여준다고 볼 수 있음

<img width="1000" alt="image" src="https://github.com/gwleeee/PaperReview/assets/83268025/b3f831de-a1e6-473e-b876-54bc4c7815aa">

- 일반적인 통합 학습(merged joints)에 비해 제안하는 학습 방법이 더 우수함을 보여줌
- 그리고 오토인코더의 입출력에 대한 loss를 이용하는 학습 방법(Consistency regularization) 방법이 제일 좋은 성능을 보여줌

<img width="500" alt="image" src="https://github.com/gwleeee/PaperReview/assets/83268025/ab520acb-fe52-4b65-ab01-600ca4c218af">

- SOTA 방법과 비교시에도 좋은 성능을 보여줌

<img width="500" alt="image" src="https://github.com/gwleeee/PaperReview/assets/83268025/8e1aee3c-042b-4e46-a222-a6632c3a60cc">

- 오토인코더의 잠재 키포인트 개수에 따른 성능
